### ISSUE TITLE
Upgrading to 8.0.0 without any other changes updates or forces replacement of several resources

### ISSUE THREAD

**Issue by nbaju1** [jfrog/terraform-provider-artifactory#744]
**Describe the bug**
Basically the title. No other changes were made except upgrading provider version (from 7.2.1). 
The complete list of changes are too large to be pasted here, so will instead provide examples.
Examples (names have been changed or redacted):

```terraform
  # module.security.artifactory_group.admins will be updated in-place
  ~ resource "artifactory_group" "admins" {
      + detach_all_users = false
      ~ id               = "Admins" -> (known after apply)
        name             = "Admins"
      ~ realm            = "scim" -> "internal"
      - users_names      = [] -> null
        # (7 unchanged attributes hidden)
    }
  
  # module.security.artifactory_scoped_token.example_token must be replaced
-/+ resource "artifactory_scoped_token" "example_token" {
      ~ access_token            = (sensitive value)
      ~ expires_in              = 31536000 # forces replacement -> (known after apply)
      ~ expiry                  = 1715328105 -> (known after apply)
      + grant_type              = "client_credentials" # forces replacement
      ~ id                      = "redacted -> (known after apply)
      + include_reference_token = false # forces replacement
      ~ issued_at               = 1683792105 -> (known after apply)
      ~ issuer                  = "jfac@xxx" -> (known after apply)
      + reference_token         = (sensitive value)
      + refresh_token           = (sensitive value)
      ~ scopes                  = [ # forces replacement
          - "applied-permissions/user",
        ] -> (known after apply)
      ~ subject                 = "jfac@xxx/users/example-user" -> (known after apply)
      ~ token_type              = "Bearer" -> (known after apply)
        # (2 unchanged attributes hidden)
    }
```
Also got several errors. Examples:

```
│ Error: Failed to decode resource from state
│ 
│ Error decoding
│ "module.security.artifactory_permission_target.example_target" from
│ previous state: missing expected {
╵
╷
│ Error: Unable to Read Previously Saved State for UpgradeResourceState
│ 
│   with module.security.artifactory_permission_target.other_target,
│   on ../modules/security/main.tf line 71, in resource "artifactory_permission_target" "other_target":
│   71: resource "artifactory_permission_target" "other_target" {
│ 
│ There was an error reading the saved resource state using the current
│ resource schema.
│ 
│ If this resource state was last refreshed with Terraform CLI 0.11 and
│ earlier, it must be refreshed or applied with an older provider version
│ first. If you manually modified the resource state, you will need to
│ manually modify it to match the current resource schema. Otherwise, please
│ report this to the provider developer:
│ 
│ AttributeName("build"): invalid JSON, expected "{", got "["
```
Terraform 1.4.2
Artifactory 8.0.0

**Expected behavior**
No changes to resources.

**Comment by alexhung** [comment]
@nbaju1 Thanks for the report. This is obviously a bug and not meant to happen. The migration to new Terraform framework should not induce any changes to existing configuration.

Can you let us know if all the changes are confined to the following resource types?
- artifactory_user
- artifactory_anonymous_user
- artifactory_managed_user
- artifactory_group
- artifactory_scoped_token
- artifactory_permission_target

If you are seeking to upgrade to the latest "safe" version, we recommend 7.7.0 which is the version before the framework migration. You should have no issue upgrading to 7.7.0. If there are, please let us know in a separate issue.

**Comment by nbaju1** [comment]
> @nbaju1 Thanks for the report. This is obviously a bug and not meant to happen. The migration to new Terraform framework should not induce any changes to existing configuration.
> 
> Can you let us know if all the changes are confined to the following resource types?
> 
> * artifactory_user
> * artifactory_anonymous_user
> * artifactory_managed_user
> * artifactory_group
> * artifactory_scoped_token
> * artifactory_permission_target
> 
> If you are seeking to upgrade to the latest "safe" version, we recommend 7.7.0 which is the version before the framework migration. You should have no issue upgrading to 7.7.0. If there are, please let us know in a separate issue.

All the changes are confined to the resource types you have listed.

**Comment by EvertonSA** [comment]
I'm also facing the same, for me it's only affecting artifactory_scoped_token

**Comment by EvertonSA** [comment]
I confirm 7.7.0 works as intended

**Comment by erlarson85** [comment]
Yes I'm pinning to 7 as well for now. Is there a breaking changes list or is this possibly resolved in a 8.0.x release?

**Comment by alexhung** [comment]
@erlarson85 So the original breaking change that led to v8 is we removed `artifactory_resource_permission_target` from the provider. I'm currently working on fixing these state drift issue from <7.7.0 provider. It's likely we will bump major version again since in order to regain backward compatibility, I may have to change the schema slightly (revert to be like pre-migration).

**Comment by alexhung** [comment]
Progress update: currently I'm blocked by these 2 issues in the Terraform framework: https://github.com/hashicorp/terraform-plugin-framework/issues/510 and https://github.com/hashicorp/terraform/issues/31887

The second one is the main blocker. The code I am testing now shows 'empty' state drift from  Terraform CLI and thus unable for me to discern which attributes have the drift. All the various attempts in working around this issue have not worked thus far and I'm awaiting response from HashiCorp team to see if there *is* any solution.

**Comment by alexhung** [comment]
@nbaju1 @EvertonSA @erlarson85 ☝🏼 I'm hopeful that this will fix all the state drift issue but I'd recommend testing this out first in your dev/testing environments 😄 Once this is released, please keep us inform if you are still seeing issues.

**Comment by EvertonSA** [comment]
in ran a plan here and I get the following:

Plan: 0 to add, 100 to change, 0 to destroy.

not the 100 changes are related to this issue, but the below remains:

  # module.jfrog_project_XXXX.artifactory_scoped_token.system_user_token will be updated in-place
  ~ resource "artifactory_scoped_token" "system_user_token" {
        id                      = "c8XXXXXXXXXXXXXXXXXXXXX1a5"
      + include_reference_token = false
        # (11 unchanged attributes hidden)
    }
    
do you happen to know if  this include_reference_token set to false going to break the system_user_token?

**Comment by EvertonSA** [comment]
I also noticed that this only happened to certain resources. from my understanding, artifactory_scoped_token created after some version (will have to check which one) works, and old ones shows the diff on include_reference_token

**Comment by EvertonSA** [comment]
I saw no other option on how to proceed other than pulling the state and adjusting 

"include_reference_token": null,

to 

"include_reference_token": false

after that, plan works as expected.

**Comment by alexhung** [comment]
@EvertonSA 

> artifactory_scoped_token created after some version (will have to check which one) works, and old ones shows the diff on include_reference_token

Correct. This is mentioned in the `NOTES` section in the `CHANGELOG.md`. Unfortunately for those who created the affected resources between the version when they were updated and the version they were fixed, there would be state drifted and require manual intervention.
